cleanlab/filter.py

```diff
@@ -290,38 +281,43 @@ def get_noise_indices(
+      and the most likely mislabeled example in class labels = 1.
 
       Note
       ----
-      Only set this parameter if ``prune_method == 'prune_by_class'``
-      You may use with ``prune_method == 'prune_by_noise_rate'``, but
+      Only set this parameter if ``filter_by == 'prune_by_class'``
+      You may use with ``filter_by == 'prune_by_noise_rate'``, but
       if ``num_to_remove_per_class == k``, then either k-1, k, or k+1
       examples may be removed for any class. This is because noise rates
       are floats, and rounding may cause a one-off. If you need exactly
       'k' examples removed from every class, you should use ``'prune_by_class'``
 
-    prune_method : str (default: 'prune_by_noise_rate')
-      Possible Values: 'prune_by_class', 'prune_by_noise_rate', or 'both'.
-      Method used for pruning.
+    filter_by : str (default: 'prune_by_noise_rate')  TODO: change default to cl_off_diag?
+      Possible Values: {'prune_by_class', 'prune_by_noise_rate', 'both',
+                        'confident_learning', 'argmax_not_equal_given_label'}
+      Method used for filtering/pruning out the label issues.
       1. 'prune_by_noise_rate': works by removing examples with
       *high probability* of being mislabeled for every non-diagonal
-      in the prune_counts_matrix (see pruning.py).
+      in the prune_counts_matrix (see filter.py).
       2. 'prune_by_class': works by removing the examples with *smallest
       probability* of belonging to their given class label for every class.
       3. 'both': Finds the examples satisfying (1) AND (2) and
       removes their set conjunction.
+      4. 'confident_learning': Find examples that are confidently labeled as a class
+      that'labels different from their given label while computing the confident joint.
+      5. 'argmax_not_equal_given_label': Find examples where the argmax predictions does not match
+      the given label.
 
-    sorted_index_method : {:obj:`None`, :obj:`prob_given_label`, :obj:`normalized_margin`}
+    return_indices_ranked_by : {:obj:`None`, :obj:`prob_given_label`, :obj:`normalized_margin`}
       If None, returns a boolean mask (true if example at index is label error)
       If not None, returns an array of the label error indices
       (instead of a bool mask) where error indices are ordered by the either:
-      ``'normalized_margin' := normalized margin (p(s = k) - max(p(s != k)))``
-      ``'prob_given_label' := [psx[i][labels[i]] for i in label_errors_idx]``
+      ``'normalized_margin' := normalized margin (p(label = k) - max(p(label != k)))``
+      ``'prob_given_label' := [psx[i][labels[i]] for i in label_issues_idx]``
 
     multi_label : bool
-      If true, s should be an iterable (e.g. list) of iterables, containing a
+      If true, labels should be an iterable (e.g. list) of iterables, containing a
       list of labels for each example, instead of just a single label.
```

Comment:
clarify more what `multi_label` setting is. Also would be nice to provide toy example of what `labels` may look like when `multi_label == True`

Reply:
clarified everywhere.